### PR TITLE
BC inlet for SA solver based on local conditions at the Inlet

### PR DESCRIPTION
**Pull request by Cristopher-Morales**
## Proposed Changes
*Give a brief overview of your contribution here in a few sentences.*

Following to what was done in the pull request #1796 , we aim with this pull request to use the known fluid properties at the inlets in order to set the BC for the SA model. Also, we extent the use of the turbulent marker options for this case as It was done in the previous pull request for SST #1843. 
I can add a test case if it is needed. 

## Related Work

related to pull request #1796 , #1843 

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [X] My contribution generates no new compiler warnings (try with --warnlevel=3 when using meson).
- [X] My contribution is commented and consistent with SU2 style (https://su2code.github.io/docs_v7/Style-Guide/).
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp), if necessary.

### CODE REVIEW

File: SU2_CFD/src/solvers/CTurbSASolver.cpp [hunk 1]
```diff
@@ -547,7 +547,33 @@ void CTurbSASolver::BC_Inlet(CGeometry *geometry, CSolver **solver_container, CN
...
+           Density_Inlet = FluidModel->GetDensity();
+         }
+         const su2double Laminar_Viscosity_Inlet = FluidModel->GetLaminarViscosity();
+         const su2double* Turb_Properties = config->GetInlet_TurbVal(config->GetMarker_All_TagBound(val_marker));
+         const su2double Nu_Factor = Turb_Properties[0];
+         if (config->GetSAParsedOptions().bc) {
+           Inlet_Vars[0] = 0.005 * Nu_Factor * Laminar_Viscosity_Inlet / Density_Inlet;
+         } else {
+           Inlet_Vars[0] = Nu_Factor * Laminar_Viscosity_Inlet / Density_Inlet;
+         }
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CTurbSASolver.cpp:576]:
```suggestion
         Inlet_Vars[0] = Nu_Factor * Laminar_Viscosity_Inlet / Density_Inlet;
         if (config->GetSAParsedOptions().bc) {
           Inlet_Vars[0] *= 0.005;
         }
```

File: SU2_CFD/src/solvers/CTurbSASolver.cpp [hunk 2]
```diff
@@ -547,7 +547,33 @@ void CTurbSASolver::BC_Inlet(CGeometry *geometry, CSolver **solver_container, CN
       su2double Inlet_Vars[MAXNVAR];
       Inlet_Vars[0] = Inlet_TurbVars[val_marker][iVertex][0];
       if (config->GetInlet_Profile_From_File()) {
-        Inlet_Vars[0] *= config->GetDensity_Ref() / config->GetViscosity_Ref();
+         Inlet_Vars[0] *= config->GetDensity_Ref() / config->GetViscosity_Ref();
+      } else {
+         /*--- Obtain fluid model for computing the  kine and omega to impose at the inlet boundary. ---*/
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CTurbSASolver.cpp:552]:
```suggestion
         /*--- Obtain fluid model for computing the nu tilde to impose at the inlet boundary. ---*/
```

**Review by pcarruscag — APPROVED** [review]
LGTM

**Review by pcarruscag — APPROVED** [review]
Check if you need to update the config template or any doc on the website.
It would also be good to modify an existing test case to exercise this feature.

**Comment by Cristopher-Morales** [comment]
> Check if you need to update the config template or any doc on the website. It would also be good to modify an existing test case to exercise this feature.

Thank you so much for your comment. Yes, I will update the website with this additional feature. I will modify one of the existing test cases in this pull request.

**Pull request merged**